### PR TITLE
tests

### PR DESCRIPTION
**Pull request by pawsko**

### CODE REVIEW

**Review by brabram — CHANGES_REQUESTED** [review]
I've checked category/rental - if similar code is in tool/user then comments are also valid for those tests :) ask in case of any doubts

File: src/test/java/pl/pawsko/toolroom/category/CategoryControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+        CategoryDtoResponse category2 = new CategoryDtoResponse();
+        category1.setCategoryName("Castorama");
+        category2.setCategoryName("Obi");
+        List<CategoryDtoResponse> allCategories = Arrays.asList(category1, category2);
+
+        given(categoryService.getAllCategories()).willReturn(allCategories);
+
+        mockMvc.perform(get("/api/category")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andDo(print())
```
[the remark this answers is not in the document]
Comment (pawsko) [reply src/test/java/pl/pawsko/toolroom/category/CategoryControllerTest.java:64]: ok

File: src/test/java/pl/pawsko/toolroom/category/CategoryControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+    @Test
+    void whenGetCategoryById_thenReturnValidRequest() throws Exception {
+        CategoryDtoResponse categoryRes = new CategoryDtoResponse();
+        categoryRes.setId(1L);
+        categoryRes.setCategoryName("Castorama");
+
+        given(categoryService.getCategoryById(anyLong())).willReturn(Optional.of(categoryRes));
+
+        mockMvc.perform(get("/api/category/{id}", anyLong())
```
[the remark this answers is not in the document]
Comment (pawsko) [reply src/test/java/pl/pawsko/toolroom/category/CategoryControllerTest.java:81]: OK

File: src/test/java/pl/pawsko/toolroom/category/CategoryControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+                .andDo(print())
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.categoryName", is(categoryRes.getCategoryName())));
+
+        verify(categoryService).getCategoryById(anyLong());
+    }
+
+    @Test
+    void whenGetByIdButBadType_thenReturnBadRequest() throws Exception {
+        mockMvc.perform(get("/api/category/{id}", "stringInsteadLong")
```
[the remark this answers is not in the document]
Comment (pawsko) [reply src/test/java/pl/pawsko/toolroom/category/CategoryControllerTest.java:92]: OK

File: src/test/java/pl/pawsko/toolroom/category/CategoryControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(status().isNotFound());
+
+        verify(categoryService).getCategoryById(anyLong());
+    }
+
+    @Test
+    void whenSaveCategory_thenReturnJson() {
+        CategoryDtoRequest categoryReq = new CategoryDtoRequest();
```
[the remark this answers is not in the document]
Comment (pawsko) [reply src/test/java/pl/pawsko/toolroom/category/CategoryControllerTest.java:115]: fixed

File: src/test/java/pl/pawsko/toolroom/category/CategoryControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+        given(categoryService.saveCategory(categoryReq)).willReturn(categoryRes);
+
+        try {
+            mockMvc.perform(post("/api/category")
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .content(mapper.writeValueAsString(categoryReq)))
+                    .andDo(print())
+                    .andExpect(status().isCreated())
+                    .andExpect(header().string(HttpHeaders.LOCATION, "http://localhost/api/category/1"))
+                    .andExpect(jsonPath("$.categoryName", is(categoryReq.getCategoryName())));
```
[the remark this answers is not in the document]
Comment (pawsko) [reply src/test/java/pl/pawsko/toolroom/category/CategoryControllerTest.java:130]: I got acquainted

File: src/test/java/pl/pawsko/toolroom/category/CategoryDtoMapperTest.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+class CategoryDtoMapperTest {
+
+    @Test
+    void testMapFromCategoryToCategoryDtoResponse() {
+        CategoryDtoMapper categoryDtoMapper = new CategoryDtoMapper();
```
[the remark this answers is not in the document]
Comment (pawsko) [reply src/test/java/pl/pawsko/toolroom/category/CategoryDtoMapperTest.java:12]: fixed

File: src/test/java/pl/pawsko/toolroom/category/CategoryServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+        categoryRes2.setCategoryName("Obi");
+        categoryReq1 = new CategoryDtoRequest();
+        categoryReq1.setCategoryName("Castorama");
+    }
+
+    @Test
+    void shouldReturnTwoCategories() {
+        List<Category> categories = List.of(category1, category2);
+        List<CategoryDtoResponse> categoriesDtoRes = List.of(categoryRes1, categoryRes2);
+
```
[the remark this answers is not in the document]
Comment (pawsko) [reply src/test/java/pl/pawsko/toolroom/category/CategoryServiceTest.java:57]: fixed

File: src/test/java/pl/pawsko/toolroom/rental/RentalControllerTest.java [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+    @Test
+    void whenSaveRental_thenReturnJson() {
+        given(rentalService.saveRental(eq(rentalReq1))).willReturn(rentalRes1);
+        given(rentalRepository.existsById(anyLong())).willReturn(true);
+        given(toolRepository.existsById(anyLong())).willReturn(true);
+        given(userRepository.existsById(anyLong())).willReturn(true);
+        mapper.findAndRegisterModules();
+
+        try {
+            mockMvc.perform(post("/api/rental")
```
[the remark this answers is not in the document]
Comment (pawsko) [reply src/test/java/pl/pawsko/toolroom/rental/RentalControllerTest.java:185]: I got acquainted

File: src/test/java/pl/pawsko/toolroom/rental/RentalControllerTest.java [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+                            .contentType(MediaType.APPLICATION_JSON)
+                            .content(mapper.writeValueAsString(rentalReq1)))
+                    .andDo(print())
+                    .andExpect(status().isCreated())
+                    .andExpect(header().string(HttpHeaders.LOCATION, "http://localhost/api/rental/1"))
```
[the remark this answers is not in the document]
Comment (pawsko) [reply src/test/java/pl/pawsko/toolroom/rental/RentalControllerTest.java:190]: I got acquainted

File: src/test/java/pl/pawsko/toolroom/rental/RentalControllerTest.java [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+    @Test
+    void whenReplaceRental_thenStatusOk() throws Exception {
+        given(rentalService.replaceRental(anyLong(), eq(rentalReq1))).willReturn(Optional.of(rentalRes1));
+        given(rentalRepository.existsById(anyLong())).willReturn(true);
+        given(toolRepository.existsById(anyLong())).willReturn(true);
+        given(userRepository.existsById(anyLong())).willReturn(true);
+
+        mapper.findAndRegisterModules();
+
+        mockMvc.perform(put("/api/rental/{id}", 100L)
```
[the remark this answers is not in the document]
Comment (pawsko) [reply src/test/java/pl/pawsko/toolroom/rental/RentalControllerTest.java:208]: I got acquainted

File: src/test/java/pl/pawsko/toolroom/rental/RentalControllerTest.java [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+                LocalDateTime.of(2020, 1, 10, 10, 10, 12),
+                LocalDateTime.of(2020, 2, 10, 15, 39, 13),
+                "Not OK");
+    }
+
+    @Test
+    void whenGetAllRentals_thenReturnJsonArray() throws Exception {
+        List<RentalDtoResponse> allRentals = Arrays.asList(rentalRes1, rentalRes2);
+        given(rentalService.getAllRentals()).willReturn(allRentals);
+        System.out.println(rentalRes1.getRented());
```
[the remark this answers is not in the document]
Comment (pawsko) [reply src/test/java/pl/pawsko/toolroom/rental/RentalControllerTest.java:128]: removed println

File: src/test/java/pl/pawsko/toolroom/rental/RentalControllerTest.java [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+    }
+
+    @Test
+    void whenReplaceRentalWithNoTExistedUser_thenReturnBadRequest() throws Exception {
+        given(rentalService.replaceRental(anyLong(), eq(rentalReq1))).willReturn(Optional.empty());
+        given(rentalRepository.existsById(anyLong())).willReturn(true);
+        given(toolRepository.existsById(anyLong())).willReturn(true);
+        given(userRepository.existsById(anyLong())).willReturn(false);
+
+        mapper.findAndRegisterModules();
```
[the remark this answers is not in the document]
Comment (pawsko) [reply src/test/java/pl/pawsko/toolroom/rental/RentalControllerTest.java:260]: moved

File: src/test/java/pl/pawsko/toolroom/rental/RentalDtoMapperTest.java [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+
+        tool = new Tool(1L, "Drill", "Dr100", manufacturer, category,
+                powerType, status, 10, location, user,
+                LocalDateTime.of(2020, 5, 15, 12, 12),
+                null);
+    }
+
+    @Test
+    void testMapFromRentalToRentalDtoResponse() {
+        Rental rental = new Rental(1L,
```
[the remark this answers is not in the document]
Comment (pawsko) [reply src/test/java/pl/pawsko/toolroom/rental/RentalDtoMapperTest.java:73]: OK

**Review by brabram — APPROVED** [review]

**Pull request merged**